packages/cli/src/util/output/progress.ts

```diff
@@ -0,0 +1,22 @@
+export interface ProgressOptions {
+  width?: number;
+  complete?: string;
+  incomplete?: string;
+}
+
+/**
+ * Returns a raw progress bar string.
+ */
+export function progress(
+  current: number,
+  total: number,
+  opts: ProgressOptions = {}
+): string {
+  const width = opts?.width ?? 20;
+  const complete = opts?.complete ?? '=';
+  const incomplete = opts?.incomplete ?? '_';
+  const unit = total / width;
+  let pos = unit === 0 ? width : Math.floor(current / unit);
```

Comment:
**suggestion (non-blocking):** validate `total > 0`

If `unit` is `0`, that means `total` must have been `0`, which seems like an invalid case to call this function. Am I missing something?

If that's the case, I think we should validate that total is `> 0` at the top of this function and maybe debug log an error message and return an empty progress bar. The call site can decide what to do with an empty result, like showing a generic "Downloading..." message.

While we're validating inputs, asserting that `current >= 0` is also a good idea.

Reply:
If we do this:

```suggestion
  let pos = Math.floor(current / unit);
```

All the tests still pass. What's the intent behind the `unit === 0` check? If this function is called with a `total` of `0`, the result is empty string.

Should we make this change and add validations?

```
if (total <= 0 || current < 0) {
  return null;
} 
```